vNext/extensions/applicationinsights-dependencies-js/src/ajax.ts

```diff
@@ -492,25 +482,39 @@ export class AjaxMonitor implements ITelemetryPlugin, IDependenciesPlugin, IInst
                 // not using original request headers will result in them being lost
                 init.headers = new Headers(init.headers || (input instanceof Request ? (input.headers || {}) : {}));
-                init.headers.set(RequestHeaders.requestIdHeader, ajaxData.id);
-                if (this._config.enableRequestHeaderTracking) {
-                    ajaxData.requestHeaders[RequestHeaders.requestIdHeader] = ajaxData.id;
+                if (this._config.distributedTracingMode === DistributedTracingModes.AI || this._config.distributedTracingMode === DistributedTracingModes.AI_AND_W3C) {
+                    var id = "|" + ajaxData.traceID + "." + ajaxData.spanID;
+                    init.headers.set(RequestHeaders.requestIdHeader, id);
+                    if (this._config.enableRequestHeaderTracking) {
+                        ajaxData.requestHeaders[RequestHeaders.requestIdHeader] = id;
+                    }
                 }
                 let appId: string = this._config.appId || this._context.appId();
                 if (appId) {
                     init.headers.set(RequestHeaders.requestContextHeader, RequestHeaders.requestContextAppIdFormat + appId);
-                    ajaxData.requestHeaders[RequestHeaders.requestContextHeader] = RequestHeaders.requestContextAppIdFormat + appId;
+                    if (this._config.enableRequestHeaderTracking) {
+                        ajaxData.requestHeaders[RequestHeaders.requestContextHeader] = RequestHeaders.requestContextAppIdFormat + appId;
+                    }
+                }
+                if (this._config.distributedTracingMode === DistributedTracingModes.AI_AND_W3C || this._config.distributedTracingMode === DistributedTracingModes.W3C) {
+                    var traceparent = new Traceparent(ajaxData.traceID, ajaxData.spanID);
+                    init.headers.set(RequestHeaders.traceParentHeader, traceparent.toString());
+                    if (this._config.enableRequestHeaderTracking) {
+                        ajaxData.requestHeaders[RequestHeaders.traceParentHeader] = traceparent.toString();
+                    }
                 }
-
                 return init;
             }
             return init;
         } else if (xhr) { // XHR
             if (this.currentWindowHost && CorrelationIdHelper.canIncludeCorrelationHeader(this._config, xhr.ajaxData.getAbsoluteUrl(),
                 this.currentWindowHost)) {
-                xhr.setRequestHeader(RequestHeaders.requestIdHeader, xhr.ajaxData.id);
-                if (this._config.enableRequestHeaderTracking) {
-                    xhr.ajaxData.requestHeaders[RequestHeaders.requestIdHeader] = xhr.ajaxData.id;
+                if (this._config.distributedTracingMode === DistributedTracingModes.AI || this._config.distributedTracingMode === DistributedTracingModes.AI_AND_W3C) {
```

Comment:
What is this block for? It looks duplicated from L485

Reply:
this is for xhr request, L485 is for fetch request